scripts/idl/matter_grammar.lark

```diff
@@ -14,34 +14,42 @@ event_access: "access" "(" ("read" ":" access_privilege)? ")"
 
 event_with_access: "event" event_access? id
 
-event: event_priority event_with_access "=" number "{" struct_field* "}"
+event: event_priority event_with_access "=" number "{" (struct_field ";")* "}"
 
 ?event_priority: "critical"i -> critical_priority
                | "info"i     -> info_priority
                | "debug"i    -> debug_priority
 
 attribute_access_entry: attribute_access_type ":" access_privilege
 
-attribute_access: "access" "(" (attribute_access_entry ("," attribute_access_entry)* )? ")"
+default_value: "default"i "=" (number | ESCAPED_STRING)
```

Comment:
This is a sharp edge when not using a scalar

Reply:
Also, number should allow floats

Reply:
I don't think I will ever add non-scalar support as some sort of json-value or equivalent would be tough. I do not intend to implement a full parser for attribute values. It is probably possible but would be very painful. Code can initialize complex values.

Reply:
My immediate concern is generally to have `reviewable zap` ... so if zap does not support fancy defaults, I would not either as a start. Would be nice to have for the future, but for that I would imagine we would want either zap support (does not sound as easy) or switch to only using IDLs